flopy/discretization/modeltime.py

```diff
@@ -71,21 +226,373 @@ def totim(self):
+            2: "minutes",
+            3: "hours",
+            4: "days",
+            5: "years",
+        }
+        valid_units_list = list(valid_units.values())
+        valid_unit = None
+
+        if isinstance(units, int):
+            # map to pre-mf6 conventions
+            if 0 <= units <= 5:
+                valid_unit = valid_units[units]
+            else:
+                raise ValueError("Integer units should be between 0 - 5")
+        else:
+            units = units.lower()
+            if len(units) == 1:
+                for vu in valid_units_list:
+                    if vu.startswith(units):
+                        valid_unit = vu
+                        break
+            else:
+                scores = []
+                for vu in valid_units_list:
+                    score = SequenceMatcher(None, vu, units).ratio()
+                    scores.append(score)
+
+                uidx = scores.index(max(scores))
+                valid_unit = valid_units_list[uidx]
+
+        if valid_unit is None:
+            raise ValueError(f"Could not determine time units from user input {units}")
+
+        return valid_unit
+
+    @staticmethod
+    def _get_datetime_string_format(str_datetime):
+        """
+        Method to parse a limited number string representations of datetime
+        formats. Currently supported string formats for date time combinations
+        are....
+
+        Parameters
+        ----------
+        str_datetime : str
+            string representation of date time. See the
+            ModelTime.datetime_from_user_input documentation for supported
+            formats
+
+        Returns
+        -------
+            datetime.datetime object
+        """
+        str_datetime = str_datetime.strip().lower()
+        if "/" in str_datetime:
+            dsep = "/"
+        elif "-" in str_datetime:
+            dsep = "-"
+        else:
+            raise ValueError(
+                "Seperator type for date part of date time representation "
+                "not recognized, supported date seperator types include '/' "
+                "and '-'"
+            )
+
+        # check for time component
+        if "t" in str_datetime:
+            dtsep = "t"
+        elif " " in str_datetime:
+            dtsep = " "
+        else:
+            dtsep = None
+
+        # check if year first (yr, month, day) combo...
+        year_first = False
+        tmp = str_datetime.split(dsep)[0]
+        if len(tmp) == 4:
+            year_first = True
+
+        if dtsep is not None:
+            if year_first:
+                str_rep = f"%Y{dsep}%m{dsep}%d{dtsep}%H:%M:%S"
+            else:
+                str_rep = f"%m{dsep}%d{dsep}%Y{dtsep}%H:%M:%S"
+
+        else:
+            if year_first:
+                str_rep = f"%Y{dsep}%m{dsep}%d"
+            else:
+                str_rep = f"%m{dsep}%d{dsep}%Y"
+
+        return str_rep
+
+    @staticmethod
+    def datetime_from_user_input(datetime_obj):
```

Comment:
I think we should consider it, but it would need to be a core dependency because ModelTime is constructed much like the Grid classes on modflow model objects.

Reply:
renamed this to `parse_datetime`